include/ghex/unstructured/communication_object_ipr.hpp

```diff
@@ -410,23 +417,17 @@ namespace gridtools {
 
                     template<typename Arch,
                              typename ValueType,
-                             typename BufferType,
                              typename Memory,
                              typename Halos,
-                             typename Function,
-                             typename DeviceIdType,
-                             typename Pool,
                              typename Field = void>
-                    void set_recv_size(Memory& memory,
+                    void set_recv_size(Memory& memory, // reference not strictly needed here
```

Comment:
why is this comment useful? could you expand it?

Reply:
as before

Reply:
Actually not, sorry. Here memory is not a pointer as before, and represents the recv memory. Reference is needed. I pushed the updated version.